pkg/serverless/trace/propagation/carriers.go

```diff
@@ -253,6 +253,16 @@ func headersCarrier(hdrs map[string]string) (tracer.TextMapReader, error) {
 	return tracer.TextMapCarrier(hdrs), nil
 }
 
+// multiValueHeadersCarrier returns the tracer.TextMapReader used to extract trace
+// context from a MultiValueHeaders field of form map[string][]string.
+func multiValueHeadersCarrier(hdrs map[string][]string) (tracer.TextMapReader, error) {
```

Comment:
I suspect you can probably use https://pkg.go.dev/github.com/DataDog/dd-trace-go@v1.24.1/ddtrace/tracer#HTTPHeadersCarrier which won't require you to reinvent the wheel or do extra allocations.

Reply:
Thank you for your suggestion!

I've made the changes in the commit 270eb93324937909481c99c7ef943510e129237d to use HTTPHeadersCarrier from dd-trace-go.
I've removed the multiValueHeadersCarrier function and its associated tests, while ensuring that the final context extraction behavior remains covered by the existing tests.
https://github.com/DataDog/datadog-agent/blob/6eda6791cf50470472e9a3616360d2c33be7c2a6/pkg/serverless/trace/propagation/extractor_test.go#L544-L553